src/helpers/gamepad.ts

```diff
@@ -217,191 +226,35 @@ export const initGamepad = () => {
 
       const buttons = controller.buttons
       const axes = controller.axes
-      if (controller.id.match(/xbox|microsoft|02ea/i)) {
-        checkActionsForXbox(buttons, axes, index)
-      } else if (controller.id.match(/gamecube|0337/i)) {
-        checkActionsForGameCube(buttons, axes, index)
-      } else if (controller.id.match(/0ce6/i)) {
-        checkActionsForPlayStation5(buttons, axes, index)
-      } else if (controller.id.match(/PS3|PLAYSTATION|0268/i)) {
-        checkActionsForPlayStation3(buttons, axes, index)
-      } else {
-        // if not specific, fallback to the xbox layout, seems
-        // to be the most common for now and if not exact it seems
-        // to cover at least the left stick and the main 2 buttons
-        checkActionsForXbox(buttons, axes, index)
+      try {
+        if (controller.id.match(/xbox|microsoft|02ea/i)) {
+          checkXbox(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/gamecube|0337/i)) {
+          checkGameCube(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/0ce6/i)) {
+          checkPS5(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/PS3|PLAYSTATION|0268/i)) {
+          checkPS3(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/2563.*0523/i)) {
+          checkPS3Clone1(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/0079.*0006/i)) {
+          checkN64Clone1(buttons, axes, index, checkAction)
+        } else if (controller.id.match(/0583.*a009/i)) {
+          checkGenius1(buttons, axes, index, checkAction)
+        } else {
+          // if not specific, fallback to the xbox layout, seems
+          // to be the most common for now and if not exact it seems
+          // to cover at least the left stick and the main 2 buttons
+          checkXbox(buttons, axes, index, checkAction)
+        }
+      } catch (error) {
+        console.log(`Gamepad error: ${error}`)
```

Comment:
Please use our log functions. (logger.ts).

You could also write a wrapper around it.
e.g.:

```typescript
const gamepadLog = (type: 'info' | 'warning' | 'error', message: string) => {
  const extMessage = `Gamepad: ${message}`
  switch(type) {
    case 'info':
      logInfo(extMessage)
      break
   case 'warning':
      logWarning(extMessage)
      break
   case 'error':
     logError(extMessage)
     break
   default:
     break
  }
}
```

Reply:
This is actually on the frontend, will they work there as well? I think they are on the electron folder so cannot be imported here.

Reply:
True. So if we want this message on the terminal we should use ipcRedender('LogInfo' ...), else we keep it like that.